scikits/fitting/tests/test_delaunay.py

```diff
@@ -27,66 +27,43 @@ def setUp(self):
-        poly = np.array([1.0, 2.0, 1.0])
-        self.x = [np.linspace(-3, 3, 30), np.linspace(-3, 3, 30)]
-        xx1, xx0 = np.meshgrid(self.x[1], self.x[0])
-        self.y = poly[0] * xx0 * xx0 + poly[1] * xx0 * xx1 + poly[2] * xx1 * xx1
-        # Test data is uniform samples of same parabola, but ensure that samples do not fall outside training set
-        self.testx = [np.linspace(-1, 1, 8), np.linspace(-1, 1, 12)]
-        testx1, testx0 = np.meshgrid(self.testx[1], self.testx[0])
-        self.testy = poly[0] * testx0 ** 2 + poly[1] * testx0 * testx1 + poly[2] * testx1 ** 2
-        self.default_val = -100.0
-        # For some reason doesn't work for a single point - requires at least a 2x2 grid
-        self.outsidex = [np.array([100, 200]), np.array([100, 200])]
-        self.outsidey = np.tile(self.default_val, (len(self.outsidex[0]), len(self.outsidex[1])))
-
-    def test_fit_eval_nn(self):
-        """Delaunay2DGridFit: Basic function fitting and evaluation using data from known function with 'nn' gridder."""
-        interp = Delaunay2DGridFit('nn', default_val=self.default_val)
-        self.assertRaises(NotFittedError, interp, self.x)
-        self.assertRaises(ValueError, interp.fit, self.y, self.y)
+        # Test linear interpolation
+        interp = Delaunay2DScatterFit(default_val=self.default_val, interp_type='linear')
         interp.fit(self.x, self.y)
         y = interp(self.x)
         testy = interp(self.testx)
         outsidey = interp(self.outsidex)
-        assert_almost_equal(y[5:-5, 5:-5], self.y[5:-5, 5:-5], decimal=10)
-        assert_almost_equal(testy, self.testy, decimal=1)
+        assert_almost_equal(y, self.y, decimal=10)
+        assert_almost_equal(testy, self.testy, decimal=10)
         assert_almost_equal(outsidey, self.outsidey, decimal=10)
-
-    def test_fit_eval_linear(self):
-        """Delaunay2DGridFit: Basic function fitting and evaluation from known function with 'linear' gridder."""
-        interp = Delaunay2DGridFit('linear', default_val=self.default_val)
-        self.assertRaises(NotFittedError, interp, self.x)
-        self.assertRaises(ValueError, interp.fit, self.y, self.y)
+        # Test nearest-neighbour interpolation
+        interp = Delaunay2DScatterFit(default_val=self.default_val, interp_type='nearest')
         interp.fit(self.x, self.y)
         y = interp(self.x)
         testy = interp(self.testx)
         outsidey = interp(self.outsidex)
-        assert_almost_equal(y[5:-5, 5:-5], self.y[5:-5, 5:-5], decimal=10)
-        assert_almost_equal(testy, self.testy, decimal=1)
-        assert_almost_equal(outsidey, self.outsidey, decimal=10)
+        assert_almost_equal(y, self.y, decimal=10)
+        assert_almost_equal(testy, self.testy, decimal=10)
+        # Nearest-neighbour interpolation has no outside value
+        # assert_almost_equal(outsidey, self.outsidey, decimal=10)
+
```

Comment:
Not something you really need to fix in this pull request, but an interpolation test where all the values being interpolated between are equal doesn't seem particularly strong...